controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +841,142 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
+	t.Run("return false and don't make changes to ControlPlaneEndpoint if load balancer is not enabled and ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{
+					Enabled: false,
+				}, 
+				ControlPlaneEndpoint: nil,
+			},
+		}
+		got := setControlPlaneEndpoint(hetznerCluster)
```

Comment:
here for example you could leave an empty line... Please always put an empty line if the blocks of code do not directly relate to each other.

For example: Empty line before and after this line `got := setControlPlaneEndpoint(hetznerCluster)`